util/httputil/cors_test.go

```diff
@@ -48,8 +48,10 @@ func TestCORSHandler(t *testing.T) {
 	resp, err := client.Do(req)
 	require.NoError(t, err, "client get failed with unexpected error")
 
-	AccessControlAllowOrigin := resp.Header.Get("Access-Control-Allow-Origin")
+	Vary := resp.Header.Get("Vary")
+	require.Equal(t, "Origin", Vary, `expected "Vary: Origin" header`)
```

Comment:
nit: don't bother with filling the `msgAndArgs` in the checks like for the existing ones, the default error message is sufficient, those are repetitive and make the check more difficult to read.